### PR TITLE
pkg/config/legacy: add missing trace agent properties to legacy config converter.

### PR DESCRIPTION
**Pull request by gbbr**
Adds missing trace agent properties to legacy config converter. As a follow-up to #2678

### CODE REVIEW

**Comment by gbbr** [comment]
@olivielpeau can you PTAL if this is more along the lines of what you meant in the other PR? If yes, I will test this PR directly in trace and potentially merge it if afterwards if all is well with you too and the team.

File: pkg/config/legacy/converter.go [hunk 1]
```diff
@@ -132,44 +137,120 @@ func FromAgentConfig(agentConfig Config) error {
 	//Trace APM based configurations
 
 	if agentConfig["apm_enabled"] != "" {
-		if enabled, err := isAffirmative(agentConfig["apm_enabled"]); err == nil && !enabled {
+		if enabled, err := isAffirmative(agentConfig["apm_enabled"]); err == nil {
 			// apm is enabled by default, convert the config only if it was disabled
```
Comment (olivielpeau) [inline pkg/config/legacy/converter.go]: this comment is outdated

File: pkg/config/legacy/converter.go [hunk 2]
```diff
@@ -132,44 +137,120 @@ func FromAgentConfig(agentConfig Config) error {
...
 
-	if receiverPort, err := strconv.Atoi(agentConfig["receiver_port"]); err == nil {
+	if receiverPort, err := strconv.Atoi(agentConfig["trace.receiver.receiver_port"]); err == nil {
 		configConverter.Set("apm_config.receiver_port", receiverPort)
 	}
 
 	if agentConfig["non_local_traffic"] != "" {
-		if enabled, err := isAffirmative(agentConfig["non_local_traffic"]); err == nil && enabled {
+		if enabled, err := isAffirmative(agentConfig["non_local_traffic"]); err == nil {
 			// trace-agent listen locally by default, convert the config only if configured to listen to more
```
Comment (olivielpeau) [inline pkg/config/legacy/converter.go]: outdated comment

File: pkg/config/legacy/converter.go [hunk 3]
```diff
@@ -132,44 +137,120 @@ func FromAgentConfig(agentConfig Config) error {
...
 			configConverter.Set("apm_config.apm_non_local_traffic", enabled)
 		}
 	}
 
-	if sampleRate, err := strconv.ParseFloat(agentConfig["extra_sample_rate"], 64); err == nil {
+	if sampleRate, err := strconv.ParseFloat(agentConfig["trace.sampler.extra_sample_rate"], 64); err == nil {
```
Comment (olivielpeau) [inline pkg/config/legacy/converter.go]: why are these trace options handled both here and in `extractTraceAgentConfig`?

**Review by olivielpeau — DISMISSED** [review]
Left a few comments, overall LGTM 👍

I'm interested in how you'll be using this in the trace-agent, please ping me when those are ready :)

File: pkg/config/legacy/converter.go (hunk 3 above)
Comment (gbbr) [reply pkg/config/legacy/converter.go]: Accident, there were a couple of duplicates. I've moved everything trace related into `extractTraceAgentConfig` and removed duplicates.

**Comment by gbbr** [comment]
Thanks for looking. I'm going to try it out in trace agent and see if anything is missing. Will ping you on both PRs.

**Comment by gbbr** [comment]
@olivielpeau PTAL. Addressed your comments and submitted another minor fix. Also tried it out on trace agent. Will open PR soon once this gets merged (so I can vendor the commit inside trace-agent).

**Review by olivielpeau — APPROVED** [review]
LGTM! 👍 

I'll let you get approval from the Agent guild as well ;)

Had a cursory look at the trace-agent PR as well and it also looks good to me

**Review by mfpierre — APPROVED** [review]
LGTM 👍

**Pull request merged**